pkgs/development/compilers/llvm/7/default.nix

```diff
@@ -23,6 +23,7 @@ let
       ln -s "${cc}/lib/clang/${release_version}/include" "$rsrc"
       ln -s "${targetLlvmLibraries.compiler-rt.out}/lib" "$rsrc/lib"
       echo "-resource-dir=$rsrc" >> $out/nix-support/cc-cflags
+      echo "--sysroot=${tools.clang-unwrapped.gcc}" >> $out/nix-support/cc-cflags
```

Comment:
`clang-unwrapped.gcc` is only valid linux with a gcc compiler if it needs to point to that then has to be conditionalized properly and same with the line below actually https://github.com/NixOS/nixpkgs/blob/cd8c1a40536886114c417d5cab357e059ac41505/pkgs/build-support/cc-wrapper/default.nix#L47-L48

How does this relate to the `-isystem` flags set in the stdenv?

Reply:
`-i` is just for header files, and the sysroot doesn't affect paths given with absolute paths, just the search path for libraries given by name and toolchain binaries called during compilation (e.g. `ld`).  I don't think these things interact.

Reply:
I'm a bit suspicious of these flags because I think they're included even if you want to use libc++, but I think that's the topic of a separate PR.

Reply:
It's the default, the variables are configured by the libc++ / libstdcxxHook hooks instead. I'm pretty sure we could actually get rid of this attribute now.

However doesn't this change have the problem you mentioned and break `libcxxStdenv` then? Whether libstdc++ or libc++ should be used and what version is determined by the `extraPackages` added to the cc-wrapper. In it's current form, if I'm not mistaken, this change would result in llvmPackages_7.stdenv using a --sysroot that points to llvmPackages_5.clang-unwrapped (default stdenv on darwin) which doesn't seem right to me.

https://github.com/NixOS/nixpkgs/blob/ec3e81beafacce0088baf50c116a26460000c012/pkgs/development/compilers/llvm/7/default.nix#L54
https://github.com/NixOS/nixpkgs/blob/ec3e81beafacce0088baf50c116a26460000c012/pkgs/development/compilers/llvm/7/default.nix#L63-L64

Reply:
I've tested this (on Linux) and I don't believe it breaks `libcxxStdlib`.  It puts `libstdc++` on the library search path to be sure, but ultimately which stdlib is linked in is controlled by `-stdlib` as it's always been.

I'm less sure about Darwin and I don't have a system to test on, but on Linux `llvmPackages_7.stdenv` still gets the gcc sysroot.  Does Darwin use `libcxxStdenv` by default?  If that's so then I think it's also fine.  I only see this breaking if Darwin's `stdenv` expects to find a gcc installation relative to the Clang `outPath`, which I believe is the default sysroot.

Reply:
Currently `libstdc++` is given by the sysroot; if we want to change that behaviour maybe it should go in a separate PR?